daphne/src/testing.rs

```diff
@@ -272,3 +349,103 @@ impl DapLeader<BearerToken> for MockAggregator {
         unreachable!("not implemented");
     }
 }
+
+/// Information associated to a certain helper state for a given task ID and aggregate job ID.
+#[derive(Clone, Eq, Hash, PartialEq, Deserialize, Serialize)]
+pub(crate) struct HelperStateId {
+    task_id: Id,
+    agg_job_id: Id,
+}
+
+/// Information associated to a certain report for a given task and nonce to decide which bucket it would be put into.
+#[derive(Clone, Eq, Hash, PartialEq, Deserialize, Serialize)]
+pub(crate) struct BucketInfo {
+    task_id: Id,
+    window: u64,
+}
+
+impl BucketInfo {
+    pub fn new(task_config: &DapTaskConfig, task_id: &Id, nonce: &Nonce) -> Self {
+        let window = nonce.time - (nonce.time % task_config.min_batch_duration);
+
+        Self {
+            task_id: task_id.clone(),
+            window,
+        }
+    }
+}
+
+impl From<DapError> for TransitionFailure {
+    fn from(_e: DapError) -> TransitionFailure {
+        TransitionFailure::ReportDropped
+    }
+}
+
+// TODO(nakatsuka-y) remove dead_code once all functions are used
+#[allow(dead_code)]
+pub(crate) struct ReportStore {
+    pending: Vec<Report>,
+    processed: HashSet<Nonce>,
+    collected: bool,
+}
+
+macro_rules! checked_process {
```

Comment:
For readability, I would suggest replacing this macro with a method on `ReportStore`.

Reply:
Agreed. Changed `checked_process` into struct method.